streams/src/test/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessorTest.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Suppress;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.Change;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.apache.kafka.streams.processor.MockProcessorContext;
+import org.apache.kafka.test.MockInternalProcessorContext;
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.Test;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Random;
+
+import static java.time.Duration.ZERO;
+import static org.apache.kafka.streams.kstream.Suppress.BufferConfig.withBufferFullStrategy;
+import static org.apache.kafka.streams.kstream.Suppress.BufferFullStrategy.SHUT_DOWN;
+import static org.apache.kafka.streams.kstream.Suppress.IntermediateSuppression.withEmitAfter;
+import static org.apache.kafka.streams.kstream.Suppress.emitFinalResultsOnly;
+import static org.apache.kafka.streams.kstream.Suppress.intermediateEvents;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.fail;
+
+@SuppressWarnings("PointlessArithmeticExpression")
+public class KTableSuppressProcessorTest {
+    private static final Random RANDOM;
+
+    static {
+        final long seed = new Random().nextLong();
+        System.out.println(KTableSuppressProcessorTest.class + ": seed: [" + seed + "]");
```

Comment:
is this intentional or left over debugging?

Reply:
It's intentional. We first randomly generate a seed, then we create a (pseudo)Random for test data generation from the seed. If the test fails, we can deterministicaly reproduce the (pseudorandom) test exactly, but only if we know the seed.

That said, I want to play around with it some more, and see if I can get it to only print the seed if there are test failures. Also, there's currently no option to run the tests with the seed, but you could always just drop in the literal when you're debugging locally.

Reply:
Hmm, I don't know if I want to spend a bunch of time of this idea. I think what I'll do is switch to a fixed seed, so that we get deterministic testing while keeping the statement that there's nothing special about the values we're testing with.